src/edges_analysis/averaging/freqbin.py

```diff
@@ -157,7 +157,11 @@ def gauss_smooth(
         nsamples = _nsamples[..., decimate_at::decimate]
 
     if use_residuals:
+        #model_mask = np.where(np.isnan(dd), 0, data.model)
+        #models = convolve1d( data.model, window, mode="constant", cval=0)[..., decimate_at::decimate]
+        #models[~mask] /= nsamples[~mask]
```

Comment:
Hmmm, can you convince me that not doing this convolution is the right thing to do here? I can't remember the reasoning for this (obviously at some point I thought that doing the convolution on the model was correct). Do we have a test we can perform?

Reply:
We can ignore this PR, I think. I _added_ the convolution step, which the original code doesn't have. I was seeing spikes after the Gauss Step that were technically removed by the RFI step before. Eventually, I figured out the issue was just me not applying the flags correctly while plotting.